Don't try to start Chrome in --machine mode unless --launch-browser is supplied

DanTup opened the pull request:
\+ don't crash server if Chrome can't be launched.

Fixes #2405.
Fixes #1936.

@devoncarew @jacob314 

I had to tweak the update_version script to make it work properly (it seemed to not handle the `+` so I ended up with the existing `+3` tagged onto the end of the new version supplied. Also unsure if the change should've gone in the server changelog or devtools (lmk if I should move it to server).

devoncarew commented on packages/devtools_server/lib/src/server.dart:
```diff
@@ -74,7 +74,9 @@ Future<void> _serveDevToolsWithArgs(
   final bool machineMode = args[argMachine];
-  final bool launchBrowser = args[argLaunchBrowser];
+  // launchBrowser defaults based on machine-mode if not explicitly supplied.
+  final bool launchBrowser =
```
👍

devoncarew reviewed (approved):
lgtm, and thanks for the fix!

I'm less familiar with the update_version.sh changes; rslgtm there.

DanTup commented:
Thanks - I'll wait for @jacob314's verdict on the script changes before merging in case I've misunderstood (I'm also not familiar with it, and having to change it felt strange since DevTools releases often have a `+` in them - I may have been using it wrong!).

I tested in VS Code and I do see the Chrome window open (which is confusing, since it's just on the main connect screen) so it might be nice to do a release with this change in once merged (as far as I can tell, there have been no other changes since the last release so it would be just this).

kenzieschmoll commented:
This also fixes https://github.com/flutter/devtools/issues/1936

DanTup commented:
Aha, in that case I'm confident in the change to the script. I wasn't sure if I was doing something wrong, but apparently it wasn't just me. Thanks!
